ui/raidboss/data/06-ew/raid/p4s.ts

```diff
@@ -796,6 +849,102 @@ const triggerSet: TriggerSet<Data> = {
       netRegexJa: NetRegexes.startsUsing({ id: '6A2D', source: 'ヘスペロス', capture: false }),
       response: Responses.aoe(),
     },
+    {
+      id: 'P4S Act',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['6A0C', '6EB[4-7]', '6A36'], source: 'Hesperos' }),
+      netRegexDe: NetRegexes.startsUsing({ id: ['6A0C', '6EB[4-7]', '6A36'], source: 'Hesperos' }),
+      netRegexFr: NetRegexes.startsUsing({ id: ['6A0C', '6EB[4-7]', '6A36'], source: 'Hespéros' }),
+      netRegexJa: NetRegexes.startsUsing({ id: ['6A0C', '6EB[4-7]', '6A36'], source: 'ヘスペロス' }),
+      run: (data, matches) => {
+        const actMap: { [id: string]: string } = {
+          '6A0C': '1',
+          '6EB4': '2',
+          '6EB5': '3',
+          '6EB6': '4',
+          '6EB7': 'finale',
+          '6A36': 'curtain',
+        };
+        data.act = actMap[matches.id];
+      },
+    },
+    {
+      id: 'P4S Thorns Collector',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6A0C', source: 'Hesperos' }),
+      netRegexDe: NetRegexes.startsUsing({ id: '6A0C', source: 'Hesperos' }),
+      netRegexFr: NetRegexes.startsUsing({ id: '6A0C', source: 'Hespéros' }),
+      netRegexJa: NetRegexes.startsUsing({ id: '6A0C', source: 'ヘスペロス' }),
+      promise: async (data, matches, _output) => {
+        // Collect all Hesperos entities up front
+        let combatantName = null;
+        combatantName = matches.source;
+
+        let combatantData = null;
+        if (combatantName) {
+          combatantData = await callOverlayHandler({
+            call: 'getCombatants',
+            names: [combatantName],
+          });
+        }
+
+        // if we could not retrieve combatant data, the
+        // trigger will not work, so just resume promise here
+        if (combatantData === null) {
+          console.error(`Hesperos: null data`);
+          return;
+        }
+        if (!combatantData.combatants) {
+          console.error(`Hesperos: null combatants`);
+          return;
+        }
+        if (combatantData.combatants.length < 39) {
```

Comment:
In raid emulator, there's only ~22 on one log that I have.  Do you really need 39?

Reply:
Every log I have has 39. It needs the last 8 npcs that get loaded, at least for just the current thorns callouts. I hadn't seen a 22 combatant log, but I guess they exist then?

Reply:
Also of note, will need to rework the other thorns triggers when the array of combatants is not 39. It could be done to just have the value of data.thornIds be the last 8 in the list instead of all Hesperos combatants, but then I am not certain where the others went if it's less than 39. Perhaps there is a locale difference in some of these Hesperos combatants that I am not seeing?

Reply:
I have added the splicing. I think it's possible to just gather all the combatants instead of Hesperos only if what happens is some of these thorns in other Locales have different names, but I think this needs to be verified in other Locales?